### PR TITLE
Fix out-of-range exception

### PR DESCRIPTION
**Pull request by chenx-dust**
When using relative topic name to subscribe, and the name is shorter than namespace, it will lead to out-of-range exception.
This PR is to fix it.

**Backport to older version needed!!!**

### CODE REVIEW

File: compressed_depth_image_transport/src/compressed_depth_publisher.cpp [hunk 1]
```diff
@@ -115,8 +115,10 @@ void CompressedDepthPublisher::advertiseImpl(
 
   // Declare Parameters
   uint ns_len = node->get_effective_namespace().length();
-  std::string param_base_name = base_topic.substr(ns_len);
-  std::replace(param_base_name.begin(), param_base_name.end(), '/', '.');
+  if (base_topic.length() >= ns_len) {
+    std::string param_base_name = base_topic.substr(ns_len);
```
Comment (ahcorde) [inline compressed_depth_image_transport/src/compressed_depth_publisher.cpp:119]: Did you already try this patch ? There is a compilation error 

```bash
14:50:25 /tmp/ws/src/image_transport_plugins/zstd_image_transport/src/zstd_publisher.cpp: In member function ‘virtual void zstd_image_transport::ZstdPublisher::advertiseImpl(rclcpp::Node*, const string&, rmw_qos_profile_t, rclcpp::PublisherOptions)’:
14:50:25 /tmp/ws/src/image_transport_plugins/zstd_image_transport/src/zstd_publisher.cpp:93:22: error: ‘param_base_name’ was not declared in this scope
14:50:25    93 |     declareParameter(param_base_name, pd);
```

`param_base_name` is defined inside the `if` scope but this variable is used in the outer scope

**Review by ahcorde — CHANGES_REQUESTED** [review]

File: compressed_depth_image_transport/src/compressed_depth_publisher.cpp (hunk 1 above)
Comment (chenx-dust) [reply compressed_depth_image_transport/src/compressed_depth_publisher.cpp:119]: I haven't tried it, sorry for that. I found this issue resulting in exception in my code, and I adopted a workaround, and then I forgot to compile and test this patch. I'll fix it soon.

**Comment by ahcorde** [comment]
friendly ping @chenx-dust

**Comment by chenx-dust** [comment]
> friendly ping @chenx-dust 

Sorry. I'm quite busy recently. If possible, please solve it for me, thank you. 🙏

I'll close this PR until I'm available to resolve it. Sorry for any inconvenience.

**Pull request closed without merging**